added tsp data in campaign details page

therocket-gr opened the pull request:
Related to #

3002-gh-tumblr/a8c-dsp

## Proposed Changes

We are adding data for the TSP blaze ad in the campaign details page

/advertising/campaigns/:campaign_id/:blog_url

the changes should look like the below SS

<img width="774" alt="Screenshot 2025-02-12 at 12 42 54 PM" src="https://github.com/user-attachments/assets/3b16fef0-0762-4f1d-94c3-402e45eb033d" />

below I am highlighting the new changes


<img width="774" alt="Screenshot 2025-02-12 at 12 42 54 PM-2" src="https://github.com/user-attachments/assets/567975da-d9a5-4aa8-94ae-c5469176281d" />

we changed some minor user displaying strings (from clicks to visitors)

and we are adding the new section that is bringing tsp related data and metrics



## Testing Instructions

if DSP PR 3141 is not yet merged, checkout this branch to your local DSP repo

1. create a Blaze campaign with TSP . (you can fake some clicks, impressions and replies in your local db)

2. bridge calypso to your DSP server and go the campaign details page

3. you should also try to have some likes, reblogs etc in your permalink so that you can see the replies (as shown above) - try to have more than 3 so you also get the show more  button and check the functionality

4 check that the permalink works 



## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

therocket-gr committed:
added tsp data in campaign details page

therocket-gr commented:
i also tested this PR in jetpack just to make sure that everything looks ok. here are the results

<img width="750" alt="Screenshot 2025-02-12 at 2 26 32 PM" src="https://github.com/user-attachments/assets/eb9e45e5-05d6-4315-a651-43f3b18fa2b6" />

sbarbosa commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -179,8 +180,20 @@ export default function CampaignItemDetails( props: Props ) {
 		conversion_rate,
 		conversion_last_currency_found,
+		tsp,
+		likes_total,
```
I mentioned this on the Backend PR, but I think we should read the likes and replies from the tsp stats object.

sbarbosa commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -827,9 +848,14 @@ export default function CampaignItemDetails( props: Props ) {
 										<div>
 											<span className="campaign-item-details__label">
-												{ translate( 'Clicks' ) }
+												{ translate( 'Visitors' ) }
```
I don't think we want to change this label. Clicks and visitors are not the same metrics. 

We currently don't have a way to track visitors, so in the last project, where we updated the campaign details page, we decided for clicks.

therocket-gr replied:
bummer.. i just checked the figma file and saw this change.. and I thought we should change it.. will revert it

therocket-gr replied:
i rolled back all changes i did to replace clicks with visitors (not its clicks again)
@sbarbosa

sbarbosa commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -1006,6 +1032,90 @@ export default function CampaignItemDetails( props: Props ) {
 											) }
 										</>
 									) }
+									{ areStatsEnabled && tsp && (
```
We don't need the check for `areStatsEnabled`. That variable checks if Jetpack has the `/stats` endpoint whitelisted. We don't use any information from that endpoint, so we should be able to render this section correctly without that extra check.

therocket-gr replied:
fixed

sbarbosa commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -1006,6 +1032,90 @@ export default function CampaignItemDetails( props: Props ) {
+														</span>
+													</span>
+												</div>
+												<div>
```
We need to comment/hide this section for now. We can't correctly track clicks using this new format.
But, I am not sure if you could run a test for this new format, and verify if clicks are getting tracked by our ad server. If not, we need to hide it for now.

Also, we shouldn't call this section "Site visits from Tumblr Post", like I mentioned before, we use clicks on the ads, not visits.

sbarbosa reviewed (commented):
Hi @therocket-gr, I tested it, and it works fine, but I found a couple of things we need to address.

therocket-gr committed:
roll back visits to clicks
change in condition to show tsp stats

therocket-gr committed:
use likes_total and replies_total from the tsp object
- comment out the clicks from tumblr post until we figure out if clicks are tracked from smart

sbarbosa reviewed (approved):
LGTM 👍

timur987 commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -1006,6 +1034,91 @@ export default function CampaignItemDetails( props: Props ) {
+														</span>
+													</span>
+												</div>
+												{ /* commenting this until we figure out if this is working properly*/ }
```
TODO would be appropriate here in order to highlight our future intention

timur987 commented on client/data/promote-post/use-promote-post-campaigns-query.ts:
```diff
@@ -43,6 +43,21 @@ export type CampaignResponse = {
+				notes: {
+					blog_name: string;
+					type: string;
+					blog_url: string;
```
Could you add `reply_text?: string;` to place actual comment?

therocket-gr replied:
added

timur987 commented on client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:
```diff
@@ -1006,6 +1034,91 @@ export default function CampaignItemDetails( props: Props ) {
+																<a href={ note.blog_url } target="_blank" rel="noopener noreferrer">
+																	@{ note.blog_name }
+																</a>
+																<br />
```
We should probably add something like the following to output actual reply test:
```js
{ note?.reply_text && note.reply_text }
```

therocket-gr replied:
added.. thanks

timur987 reviewed (approved):
Thanks @therocket-gr, overall it works great, but I want this PR to be even better.

For my case I used a post that has both like and a reply and I noticed that there'a note with type "posted" and showing it makes not sense. Also, there's no output for actual reply (but I assume it should be there).

Below is the patch for A8C DSP which:
1. Filter out "posted" type of note,
2. Adds a node "reply_text" (feel free to adjust naming) to store actual comment

(According to [Figma](https://www.figma.com/design/C8gRy5FoZtYxuF2WlvqVSs/WP-Blaze---Super-native-ad-format?node-id=42-2883&t=udneDniz3dlpEc9Q-0) we should have them)

<img width="787" alt="image" src="https://github.com/user-attachments/assets/e1e281d5-a4cb-4f9e-a152-1b979d50d8a5" />
(image from Figma)

[patch.txt](https://github.com/user-attachments/files/18785652/patch.txt)

The couple of comments below to remind of adding code to actual showing those replies.

According to figma we should also have a "See more" button, but it seems to be a completely different story.

P.S.: I don't mind releasing this as-is, but adding code to WP Calypso now looks as a time-saver for us since we have such a tight deadline.

timur987 commented:
It's about "posted" type

https://github.com/user-attachments/assets/b63aa2e3-3d46-41ef-84ca-19245cbbc667

therocket-gr committed:
added the reply text in the notes section
added some todos in the comments

timur987 reviewed (approved):
Thanks for addressing everything, @therocket-gr! 🥇

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17253003).
